server/internal/store/cluster.go

```diff
@@ -27,13 +27,33 @@ type ClusterSpec struct {
 
 // CreateCluster creates a cluster.
 func (s *S) CreateCluster(spec ClusterSpec) (*Cluster, error) {
+	var components = []string{
+		"inference-manager-engine",
+		"model-manager-loader",
+		"session-manager-agent",
+		"job-manager-dispatcher",
+	}
+
 	c := &Cluster{
 		ClusterID:       spec.ClusterID,
 		TenantID:        spec.TenantID,
 		Name:            spec.Name,
 		RegistrationKey: spec.RegistrationKey,
 	}
-	if err := s.db.Create(c).Error; err != nil {
+	if err := s.db.Transaction(func(tx *gorm.DB) error {
+		if err := s.db.Create(c).Error; err != nil {
+			return err
+		}
+		for _, name := range components {
+			if err := s.CreateClusterComponent(&ClusterComponent{
```

Comment:
We need to pass `tx` to this function to run the DB query in the same transaction.